### DISCUSSION TITLE
Scan CODE_39 EXTENDED

### DISCUSSION THREAD

**Discussion by igor-kulikov** [ericblade/quagga2#363] [Q&A]
Hi,

could you inform is it possible to easily teach library to scan Code39_Extended barcode format?

like [base ZXing library](https://zxing.github.io/zxing/apidocs/com/google/zxing/oned/Code39Reader.html) by providing configuration property USE_CODE_39_EXTENDED_MODE

thanks in advance

**Comment by ericblade** [comment]
I really don't know the answer to that -- I haven't tried to implement any new readers.  

It *looks like* it might be simple -- https://help.accusoft.com/BarcodeXpress/v13.2/BxNodeJs/code_39_extended.html indicates that the extended codes are all some double-byte version of the regular code 39 encoding.  If that's the case, can probably just hit it with the regular code 39 scanner, and do a quick pass over the results and reinterpret anything that has a 2-byte character code.  

Here is a complete chart of Code 39 codes, as well as the Code 39 extended https://www.barcodefaq.com/1d/code-39/

I would definitely welcome a patch with support and tests!

At the very least, if you could post some tests with Code 39 Extended, we could see if Code 39 will read them as something
